codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -181,8 +183,8 @@ public ChannelFuture writeData(final ChannelHandlerContext ctx, final int stream
         }
 
         // Hand control of the frame to the flow controller.
-        ChannelFuture future =
-                flowController().sendFlowControlledFrame(ctx, stream, data, padding, endOfStream, promise);
+        ChannelFuture future = flowController().sendFlowControlledData(ctx, stream,
+                new FlowControlledData(ctx, streamId, data, padding, endOfStream, promise), promise);
```

Comment:
Based on Scott's comment it seemed like the goal was to separate the responsibilities between encoder & flow-controller. The encoder knew how to write payloads and the controller would say when they could be written. In theory with this change you could drop in other flow control implementations pretty easily. This seems cleaner to me overall and it certainly made the tests cleaner I believe.